app/electron/preload.ts

```diff
@@ -4,13 +4,32 @@ import { contextBridge, ipcRenderer } from 'electron';
 contextBridge.exposeInMainWorld('desktopApi', {
   send: (channel, data) => {
     // allowed channels
-    const validChannels = ['setMenu', 'locale', 'appConfig'];
+    const validChannels = [
+      'setMenu',
+      'locale',
+      'appConfig',
+      'port-forward',
+      'port-forward-initiated',
+      'port-forward-started',
+      'port-forward-stopped',
+      'port-forward-stop',
```

Comment:
A couple of these (initiated/started) would not be needed as they are part of the app -> renderer only (receive end).
Vice-versa for the receive ones below.

I think this could be made simpler if you have just one method `port-forward` and then you have an `action` parameter passed together with any others, where you identify what's going on (also notice the case without dashes, for consistency):

```typescript
desktopApi.send('portForward', {action: 'start', port...})
```

```typescript
desktopApi.receive('portForward', ({action, port}) => ...)
```

Reply:
this is now solved by the backend api